db/migrate/20190805170949_change_columns_to_not_null.rb

```diff
@@ -0,0 +1,121 @@
+class ChangeColumnsToNotNull < ActiveRecord::Migration[5.1]
+  def change
+    change_column_null :pipeline_run_stages, :pipeline_run_id, false
+    change_column_null :pipeline_run_stages, :step_number, false
+    change_column_null :pipeline_run_stages, :job_command_func, false
+    change_column_null :pipeline_run_stages, :name, false
+
+    change_column_null :alignment_configs, :name, false
+    change_column_null :alignment_configs, :s3_nt_db_path, false
+    change_column_null :alignment_configs, :s3_nt_loc_db_path, false
+    change_column_null :alignment_configs, :s3_nr_db_path, false
+    change_column_null :alignment_configs, :s3_nr_loc_db_path, false
+    change_column_null :alignment_configs, :s3_lineage_path, false
+    change_column_null :alignment_configs, :s3_accession2taxid_path, false
+    change_column_null :alignment_configs, :s3_deuterostome_db_path, false
+    change_column_null :alignment_configs, :lineage_version, false
+
+    change_column_null :host_genomes, :name, false
+    change_column_null :host_genomes, :s3_star_index_path, false
+    change_column_null :host_genomes, :s3_bowtie2_index_path, false
+    change_column_null :host_genomes, :default_background_id, false
+
+    change_column_null :input_files, :name, false
+    change_column_null :input_files, :sample_id, false
+    change_column_null :input_files, :source_type, false
+    change_column_null :input_files, :source, false
+
+    change_column_null :metadata_fields, :display_name, false
+    change_column_null :metadata_fields, :force_options, false
+    change_column_null :metadata_fields, :is_core, false
+    change_column_null :metadata_fields, :is_default, false
+    change_column_null :metadata_fields, :is_required, false
+    change_column_null :metadata_fields, :default_for_new_host_genome, false
+
+    change_column_null :output_states, :output, false
+    change_column_null :output_states, :pipeline_run_id, false
```

Comment:
Same question about `create_output_states` happening `before_create` of `pipeline_run`